packages/core/src/codewhisperer/commands/startTransformByQ.ts

```diff
@@ -92,6 +97,64 @@ export async function processTransformFormInput(
     transformByQState.setTargetJDKVersion(toJDKVersion)
 }
 
+export async function processSQLConversionTransformFormInput(pathToProject: string, schema: string) {
+    transformByQState.setProjectName(path.basename(pathToProject))
+    transformByQState.setProjectPath(pathToProject)
+    transformByQState.setSchema(schema)
+    transformByQState.setSourceJDKVersion(JDKVersion.JDK8) // use dummy value of JDK8 so that startJob API can be called
+    // targetJDKVersion defaults to JDK17, the only supported version, which is fine
+}
+
+export async function validateSQLMetadataFile(fileContents: string, message: any) {
+    try {
+        let sctData: any = undefined
+        xml2js.parseString(fileContents, (err, result) => {
+            if (err) {
+                getLogger().error('CodeTransformation: Error parsing .sct file, not valid XML.', err)
+                throw new Error(`Invalid XML encountered`)
+            } else {
+                sctData = result
```

Comment:
this is done in a callback, is it asynchronous? yet the code just below references `sctData`...

Reply:
I see now that `xml2j` has an equivalent function which returns a `Promise`, will update